core/services/feeds/service.go

```diff
@@ -803,7 +803,7 @@ func (s *service) ApproveSpec(ctx context.Context, id int64, force bool) error {
 					}
 				}
 			case job.Workflow:
-				existingJobID, txerr = findExistingWorkflowJob(ctx, *j.WorkflowSpec, tx.jobORM)
+				existingJobID, txerr = tx.jobORM.FindJobIDByWorkflow(ctx, *j.WorkflowSpec)
```

Comment:
I wonder if this will actually find the job here tho. `FindJobIDByWorkflow` takes a wfspec and uses the `workflow_spec_id` and `workflow_spec_id` and `workflow_name`. But here, the job is a newly generated job that hasn't been persisted yet with an id of 0, and the corresponding j.WorkflowSpec would also not have an id.

Reply:
I think what we need here is probably to implement the removed function below inside the feeds/orm.go here and only use the `workflow_name` + `workflow_owner`

Reply:
Unrelated to this PR but I wonder if `FindJobIDByWorkflow` needs the rest of the arguments as well. Would the `workflow_spec_id` not be sufficient? That would mean there are multiple jobs for the same workflow spec that have different owner or name?